pkg/lockfile/parse-conan-lock.go

```diff
@@ -0,0 +1,177 @@
+	// conan v0.4- lockfiles use "graph_lock", "profile_host" and "profile_build"
+	GraphLock    ConanGraphLock `json:"graph_lock,omitempty"`
+	ProfileHost  string         `json:"profile_host,omitempty"`
+	ProfileBuild string         `json:"profile_build,omitempty"`
+	// conan v0.5+ lockfiles use "requires", "build_requires" and "python_requires"
+	Requires       []string `json:"requires,omitempty"`
+	BuildRequires  []string `json:"build_requires,omitempty"`
+	PythonRequires []string `json:"python_requires,omitempty"`
+}
+
+const ConanEcosystem Ecosystem = "conan.io"
+
+func parseConanRenference(ref string) ConanReference {
+	// very flexible format name/version[@username[/channel]][#rrev][:pkgid[#prev]][%timestamp]
+	var reference ConanReference
+
+	parts := strings.SplitN(ref, "%", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		reference.TimeStamp = parts[1]
+	}
+
+	parts = strings.SplitN(ref, ":", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		parts = strings.SplitN(parts[1], "#", 2)
+		reference.PackageID = parts[0]
+		if len(parts) == 2 {
+			reference.PackageRevision = parts[1]
+		}
+	}
+
+	parts = strings.SplitN(ref, "#", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		reference.RecipeRevision = parts[1]
+	}
+
+	parts = strings.SplitN(ref, "@", 2)
+	if len(parts) == 2 {
+		ref = parts[0]
+		UsernameChannel := parts[1]
+
+		parts = strings.SplitN(UsernameChannel, "/", 2)
+		reference.Username = parts[0]
+		if len(parts) == 2 {
+			reference.Channel = parts[1]
+		}
+	}
+
+	parts = strings.SplitN(ref, "/", 2)
+	if len(parts) == 2 {
+		reference.Name = parts[0]
+		reference.Version = parts[1]
+	} else {
+		reference.Version = ref
+	}
+	return reference
+}
+
+func parseConanV1Lock(lockfile ConanLockFile) []PackageDetails {
+	var reference ConanReference
+	packages := make([]PackageDetails, 0, len(lockfile.GraphLock.Nodes))
+
+	for _, node := range lockfile.GraphLock.Nodes {
+
+		if node.Path != "" {
+			// a local "conanfile.txt", skip
+			continue
+		}
+
+		if node.Pref != "" {
+			// old format 0.3 (conan 1.27-) lockfiles use "pref" instead of "ref"
+			reference = parseConanRenference(node.Pref)
+		} else if node.Ref != "" {
+			reference = parseConanRenference(node.Ref)
+		} else {
+			continue
+		}
+		packages = append(packages, PackageDetails{
+			Name:      reference.Name,
+			Version:   reference.Version,
+			Ecosystem: ConanEcosystem,
+			CompareAs: ConanEcosystem,
+		})
+	}
+
+	return packages
+}
+
+func parseConanRequires(packages *[]PackageDetails, requires []string) {
+
+	for _, ref := range requires {
+
+		reference := parseConanRenference(ref)
+
+		*packages = append(*packages, PackageDetails{
+			Name:      reference.Name,
+			Version:   reference.Version,
+			Ecosystem: ConanEcosystem,
+			CompareAs: ConanEcosystem,
+		})
+	}
+}
+
+func parseConanV2Lock(lockfile ConanLockFile) []PackageDetails {
+
+	packages := make([]PackageDetails, 0, len(lockfile.Requires)+len(lockfile.BuildRequires)+len(lockfile.PythonRequires))
```

Comment:
You'll need to wrap these in `uint64` to make CodeQL happy, otherwise it will flag this as a potential vulnerability since the return type of `len` technically allows negative numbers even though in practice that should never be possible.

See [here](https://github.com/google/osv-scanner/blob/main/pkg/lockfile/parse-composer-lock.go#L42-L43) for an example of how I've addressed that in the past

Reply:
thanks, I am not that familiar with Go or GoLang, so negative length or size type sounds a bit suspicious for me. fixed it.